test/test_nestedtensor.py

```diff
@@ -8816,9 +8870,12 @@ def _inp():
 
 class TestNestedInt(torch.testing._internal.common_utils.TestCase):
     def test_comparisons(self):
-        a = torch.SymInt(NestedIntNode(1, 1))
-        b = torch.SymInt(NestedIntNode(1, 1))
-        c = torch.SymInt(NestedIntNode(2, 1))
+        cache = torch.tensor(1.0)
+        cache2 = torch.tensor(1.0)
+
+        a = torch.SymInt(NestedIntNode(cache, 1))
+        b = torch.SymInt(NestedIntNode(cache, 1))
+        c = torch.SymInt(NestedIntNode(cache2, 1))
```

Comment:
so if I'm reading these assertions correctly, we still require the exact same tensor between two caches for the associated SymInts to compare equal, right (i.e. cache tensor *values* being the same is not enough)?

Reply:
The CacheTensor tensor wrapper subclass doesn't need to be the same instance, but the tensors they hold need to be the same instance.